### ISSUE TITLE
Write unit tests for the document handler

### ISSUE THREAD

**Issue by Tjzabel** [RITlug/teleirc#286]
# Summary

Unit tests do not currently exist for the Telegram document handler.

# Background

**Is your improvement related to a problem? Please describe**:

After writing this handler, it is unclear if we have handled all edge cases. We should have unit tests to thoroughly test this functionality.

**Describe the solution you'd like**:

Full set of unit tests to catch as many edge cases as we can.

**Describe alternatives you've considered**:

No alternatives.

# Details

The owner of this issue needs to think about what a user can and cannot do when uploading a file to Telegram. Then, build unit tests around the desired and undesired functionality to make sure error handling exists in the appropriate locations.

# Outcome

Unit testing for the TG document handler. Allows us to feel safe knowing this handler is thoroughly tested.

**Comment by jwflory** [comment]
_Discussed in [2020-04-11 meeting](https://bluejeans.com/s/RViMY)._

---

* Note username (and if a user does not have a username set)
* Handling captions and no captions
* Note difference between documents (anything uploaded) and photos (compressed images)
* Make sure documents work even if [optional data](https://github.com/go-telegram-bot-api/telegram-bot-api/blob/master/types.go#L330-L337) is missing

**Comment by Tjzabel** [comment]
FWIW I created a helper function to aid in getting the correct username. This is very similar to how we did this in NodeJS:
https://github.com/RITlug/teleirc/blob/tg/join-part/internal/handlers/telegram/helpers.go

This specifically constructs a full username `test (@test)`. Use the `user.String()` method on any Telegram User object in order to derive the username if present, and first name otherwise.
